src/pinint/traits.rs

```diff
@@ -0,0 +1,16 @@
+/// Implemented by types that identify pin interrupts
+///
+/// This trait is an internal implementation detail and should neither be
+/// implemented nor used outside of LPC8xx HAL. Any changes to this trait won't
+/// be considered breaking changes.
+pub trait Trait {
```

Comment:
We should probably look into sealed traits for the future (Should probably not be fixed in this PR, to have a consistent style)

Reply:
Yes, definitely.

I have been aware of that technique for a while, and the reason I haven't done it for the newer traits is just pure laziness (which isn't a valid reason for anything). I'll keep it in mind for future traits, and if I happen to do more cleanup operations, I'll fix older traits to, as I go.